posthog/models/event/events_query.py

```diff
@@ -50,24 +50,40 @@ def run_events_query(
     offset = 0 if query.offset is None else query.offset
 
     # columns & group_by
-    select_input_raw = ["*"] if len(query.select) == 0 else query.select
-    select_input: List[str] = []
+    select_input = ["*"] if len(query.select) == 0 else query.select
     person_indices: List[int] = []
-    for index, col in enumerate(select_input_raw):
+
+    select: List[ast.Alias] = []
+    group_by: List[ast.Expr] = []
+    aggregations: List[ast.Expr] = []
+
+    for index, col in enumerate(select_input):
         # Selecting a "*" expands the list of columns, resulting in a table that's not what we asked for.
         # Instead, ask for a tuple with all the columns we want. Later transform this back into a dict.
         if col == "*":
-            select_input.append(f"tuple({', '.join(SELECT_STAR_FROM_EVENTS_FIELDS)})")
-        elif col.split("--")[0].strip() == "person":
+            select_expr = f"tuple({', '.join(SELECT_STAR_FROM_EVENTS_FIELDS)})"
+        elif col.split("--")[0].split(" as ")[0].strip() == "person":
             # This will be expanded into a followup query
-            select_input.append("distinct_id")
+            select_expr = "distinct_id"
             person_indices.append(index)
         else:
-            select_input.append(col)
+            select_expr = col
+
+        expr = parse_expr(select_expr)
+        # Split between fields to 1) select by and 2) select and group by
+        if has_aggregation(expr):
+            aggregations.append(expr)
+        else:
+            group_by.append(expr)
+
+        # Make sure each column has an alias
+        if isinstance(expr, ast.Alias):
+            select.append(expr)
+        elif isinstance(expr, ast.Field):
+            select.append(ast.Alias(expr=expr, alias=expr.chain[-1]))
+        else:
+            select.append(ast.Alias(expr=expr, alias=col))
```

Comment:
This is the second fix in this file.

Sorting `order by bla as x` gives an error. Thus we must extract the expression from the alias. However there's a benefit in explicitly providing an alias for every selected column: we can use the generated AST in a subquery. I ran into it when writing the test that is/was https://github.com/PostHog/posthog/pull/15859  . Making everything an alias here also simplifies the `order by` code below.